server/src/main/java/org/apache/iotdb/db/engine/storagegroup/StorageGroupProcessor.java

```diff
@@ -537,7 +540,7 @@ public void insert(InsertPlan insertPlan) throws QueryProcessException {
       while (loc < batchInsertPlan.getRowCount()) {
         long time = batchInsertPlan.getTimes()[loc];
         long curTimePartition = StorageEngine.fromTimeToTimePartition(time);
-        results[loc] = TSStatusCode.SUCCESS_STATUS.getStatusCode();
+        results[loc] = RpcUtils.getStatus(TSStatusCode.SUCCESS_STATUS);
```

Comment:
overhead if all are successful. (too many TSStatus instances.. right?)

Reply:
Yes, I create a public static TSStatus SUCCESS_STATUS = new TSStatus(TSStatusCode.SUCCESS_STATUS.getStatusCode());
in RpcUtils and reuse this.

Reply:
If all are successful, I will return one TSStatus in TSServiceImpl line 1194

if (isAllSuccessful) {
        if (logger.isDebugEnabled()) {
          logger.debug("Insert one RowBatch successfully");
        }
        return RpcUtils.getTSBatchExecuteStatementResp(TSStatusCode.SUCCESS_STATUS);
      } else {